api/types/swarm/network.go

```diff
@@ -78,12 +78,14 @@ type Network struct {
 // NetworkSpec represents the spec of a network.
 type NetworkSpec struct {
 	Annotations
-	DriverConfiguration *Driver      `json:",omitempty"`
-	IPv6Enabled         bool         `json:",omitempty"`
-	Internal            bool         `json:",omitempty"`
-	Attachable          bool         `json:",omitempty"`
-	Ingress             bool         `json:",omitempty"`
-	IPAMOptions         *IPAMOptions `json:",omitempty"`
+	DriverConfiguration *Driver           `json:",omitempty"`
+	IPv6Enabled         bool              `json:",omitempty"`
+	Internal            bool              `json:",omitempty"`
+	Attachable          bool              `json:",omitempty"`
+	Ingress             bool              `json:",omitempty"`
+	IPAMOptions         *IPAMOptions      `json:",omitempty"`
+	Reference           *NetworkReference `json:",omitempty"`
```

Comment:
This could really be a `ConfigReference` with a `Network` as a possible reference type.